### PR TITLE
Fix #678 : Landscape onBoarding workflow

### LINKED ISSUE #678: Landscape Lowif: Onboarding Flow
Mock: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/1f86ca4e-795e-4a75-8784-fea53864b94b/RG-OB-Onboarding-1-

Also, write test cases for landscape mode.

Images link: https://drive.google.com/open?id=1BVmATijZcOvBWksXeCSism-ZQvohCkLn

### PR DESCRIPTION
**Pull request by Sarthak2601**
## Explanation

- Fixes part of #678 
- Added the svg files for the landscape view
- One of the svg file contained an image link so replaced that with path source and everything to make that image come back
- Handled the source of the slide image using the orientation configuration of android in OnboadingSlideViewModel.kt
- Changed the ScreenOrientation of the onBoarding activity in the Manifest

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by Sarthak2601** [comment]
@rt4914 PTAL :-)

File: app/src/main/java/org/oppia/app/onboarding/OnboadingSlideViewModel.kt [hunk 1]
```diff
@@ -14,6 +16,7 @@ class OnboardingSlideViewModel(val context: Context, viewPagerSlide: ViewPagerSl
   val contentDescription = ObservableField<String>(context.resources.getString(R.string.onboarding_slide_0_title))
   val title = ObservableField<String>(context.resources.getString(R.string.onboarding_slide_0_title))
   val description = ObservableField<String>(context.resources.getString(R.string.onboarding_slide_0_description))
+  val orientation = Resources.getSystem().configuration.orientation
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/onboarding/OnboadingSlideViewModel.kt]: Make this private

File: app/src/main/java/org/oppia/app/onboarding/OnboadingSlideViewModel.kt [hunk 2]
```diff
@@ -22,25 +25,29 @@ class OnboardingSlideViewModel(val context: Context, viewPagerSlide: ViewPagerSl
   private fun initializingSlide(viewPagerSlide: ViewPagerSlide) {
     when (viewPagerSlide) {
       ViewPagerSlide.SLIDE_0 -> {
-        slideImage.set(R.drawable.ic_onboarding_0)
+        if(orientation == Configuration.ORIENTATION_LANDSCAPE){slideImage.set(R.drawable.ic_landscape_onboarding_2)}
+        else if(orientation == Configuration.ORIENTATION_PORTRAIT){slideImage.set(R.drawable.ic_onboarding_0)}
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/onboarding/OnboadingSlideViewModel.kt]: Write this if else code in this format:
```
       if (orientation == Configuration.ORIENTATION_LANDSCAPE) {
          slideImage.set(R.drawable.ic_landscape_onboarding_2)
        } else if (orientation == Configuration.ORIENTATION_PORTRAIT) {
          slideImage.set(R.drawable.ic_onboarding_0)
        }
```

File: app/src/main/java/org/oppia/app/onboarding/OnboadingSlideViewModel.kt [hunk 3]
```diff
@@ -22,25 +25,29 @@ class OnboardingSlideViewModel(val context: Context, viewPagerSlide: ViewPagerSl
   private fun initializingSlide(viewPagerSlide: ViewPagerSlide) {
     when (viewPagerSlide) {
       ViewPagerSlide.SLIDE_0 -> {
-        slideImage.set(R.drawable.ic_onboarding_0)
+        if(orientation == Configuration.ORIENTATION_LANDSCAPE){slideImage.set(R.drawable.ic_landscape_onboarding_2)}
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/onboarding/OnboadingSlideViewModel.kt]: The final/image numbering is incorrect. The first slide should have 0 index, second slide should have 1 index and so on, similar to previous images. Here it should be `ic_onboarding_landscape_0`

File: app/src/main/res/layout-land/onboarding_fragment.xml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+      android:layout_height="0dp"
+      android:layout_marginBottom="0dp"
+      android:overScrollMode="never"
+      android:scrollbars="none"
+      app:layout_constraintBottom_toTopOf="@id/bottom_container"
+      app:layout_constraintEnd_toEndOf="parent"
+      app:layout_constraintStart_toStartOf="parent"
+      app:layout_constraintTop_toTopOf="parent">
+
+
```
Comment (rt4914) [inline app/src/main/res/layout-land/onboarding_fragment.xml]: Remove this extra blank lines.

File: app/src/main/res/layout-land/onboarding_fragment.xml [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      android:id="@+id/onboarding_slide_view_pager"
+      android:layout_width="match_parent"
+      android:layout_height="0dp"
+      android:layout_marginBottom="0dp"
+      android:overScrollMode="never"
+      android:scrollbars="none"
+      app:layout_constraintBottom_toTopOf="@id/bottom_container"
+      app:layout_constraintEnd_toEndOf="parent"
+      app:layout_constraintStart_toStartOf="parent"
+      app:layout_constraintTop_toTopOf="parent">
```
Comment (rt4914) [inline app/src/main/res/layout-land/onboarding_fragment.xml]: We can make this a self-closing tag 
`app:layout_constraintTop_toTopOf="parent" />`

**Review by rt4914 — CHANGES_REQUESTED** [review]
The initial structure looks correct but if you look at the mock correctly, the left side of screen contains the image background.

Mock:
Screenshot 2020-02-23 at 11 45 22 AM

App:
Screenshot 2020-02-23 at 11 45 09 AM

**Comment by Sarthak2601** [comment]
> The initial structure looks correct but if you look at the mock correctly, the left side of screen contains the image background.
> 
> Mock:
> Screenshot 2020-02-23 at 11 45 22 AM
> 
> App:
> Screenshot 2020-02-23 at 11 45 09 AM

Actually, I wanted to discuss this with you. The thing here is that we have the constraint layout with the skip button and the dots at the bottom in the fragment and the slide gets loaded in the view pager above that. As the slides load in the view pager it can't be continued till the end because of the presence of the another constraint layout.

**Comment by PrarabdhGarg** [comment]
> > The initial structure looks correct but if you look at the mock correctly, the left side of screen contains the image background.
> > Mock:
> > Screenshot 2020-02-23 at 11 45 22 AM
> > App:
> > Screenshot 2020-02-23 at 11 45 09 AM
> 
> Actually, I wanted to discuss this with you. The thing here is that we have the constraint layout with the skip button and the dots at the bottom in the fragment and the slide gets loaded in the view pager above that. As the slides load in the view pager it can't be continued till the end because of the presence of the another constraint layout.

@Sarthak2601 what you could do for that is that instead of constraining the bottom of the viewpager with the top of the bottom container, you could constraint it with the bottle of the parent. That would give it the necessary overlap required.

**Comment by Sarthak2601** [comment]
> > > The initial structure looks correct but if you look at the mock correctly, the left side of screen contains the image background.
> > > Mock:
> > > Screenshot 2020-02-23 at 11 45 22 AM
> > > App:
> > > Screenshot 2020-02-23 at 11 45 09 AM
> > 
> > 
> > Actually, I wanted to discuss this with you. The thing here is that we have the constraint layout with the skip button and the dots at the bottom in the fragment and the slide gets loaded in the view pager above that. As the slides load in the view pager it can't be continued till the end because of the presence of the another constraint layout.
> 
> @Sarthak2601 what you could do for that is that instead of constraining the bottom of the viewpager with the top of the bottom container, you could constraint it with the bottle of the parent. That would give it the necessary overlap required.

Oh yes, found that. Thank you so much @PrarabdhGarg :-)

**Comment by Sarthak2601** [comment]
Updated the positioning of the get started button in accordance with the mock. 


@rt4914 Kindly review.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@Sarthak2601 Looks good, thanks. 
Update the title of this PR: `Fix #678: Landscape onBoarding workflow`

Also, now we can focus on test cases.

Check `OnboardingFragmentTest` in this you will find a lot of test cases for portrait mode. In this file you will need add following test cases for orientation mode.

1. Rotate to orientation mode, check slide_0 title, if this is correct or not.
2. In portrait mode, move to slide_1 , change to landscape mode and now check if the title is correct or not.
3. In portrait mode, click on skip, change to landscape mode and now check the title is same as last slide title or not.


Regarding how to change to orientation landscape in a test case, you can see test cases in `AdminAuthActivityTest`

File: app/src/main/java/org/oppia/app/onboarding/OnboadingSlideViewModel.kt (hunk 2 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/onboarding/OnboadingSlideViewModel.kt]: @Sarthak2601 Once you have solved this, please reply to this comment. In this case you can simply write `Done`, ditto for other comments too.

File: app/src/main/java/org/oppia/app/onboarding/OnboadingSlideViewModel.kt (hunk 1 above)
Comment (Sarthak2601) [reply app/src/main/java/org/oppia/app/onboarding/OnboadingSlideViewModel.kt]: Done.

File: app/src/main/java/org/oppia/app/onboarding/OnboadingSlideViewModel.kt (hunk 2 above)
Comment (Sarthak2601) [reply app/src/main/java/org/oppia/app/onboarding/OnboadingSlideViewModel.kt]: Okay 👍
Done :-)

File: app/src/main/java/org/oppia/app/onboarding/OnboadingSlideViewModel.kt (hunk 3 above)
Comment (Sarthak2601) [reply app/src/main/java/org/oppia/app/onboarding/OnboadingSlideViewModel.kt]: Done

File: app/src/main/res/layout-land/onboarding_fragment.xml (hunk 4 above)
Comment (Sarthak2601) [reply app/src/main/res/layout-land/onboarding_fragment.xml]: Done 👍

File: app/src/main/res/layout-land/onboarding_fragment.xml (hunk 5 above)
Comment (Sarthak2601) [reply app/src/main/res/layout-land/onboarding_fragment.xml]: Done 👍

**Comment by Sarthak2601** [comment]
> @Sarthak2601 Looks good, thanks.
> Update the title of this PR: `Fix #678: Landscape onBoarding workflow`
> 
> Also, now we can focus on test cases.
> 
> Check `OnboardingFragmentTest` in this you will find a lot of test cases for portrait mode. In this file you will need add following test cases for orientation mode.
> 
> 1. Rotate to orientation mode, check slide_0 title, if this is correct or not.
> 2. In portrait mode, move to slide_1 , change to landscape mode and now check if the title is correct or not.
> 3. In portrait mode, click on skip, change to landscape mode and now check the title is same as last slide title or not.
> 
> Regarding how to change to orientation landscape in a test case, you can see test cases in `AdminAuthActivityTest`

Okay,  I will work on it 👍
Thank you so much for the guidance :-)

**Comment by Sarthak2601** [comment]
> The initial structure looks correct but if you look at the mock correctly, the left side of screen contains the image background.
> 
> Mock:
> Screenshot 2020-02-23 at 11 45 22 AM
> 
> App:
> Screenshot 2020-02-23 at 11 45 09 AM

Done 👍

**Comment by Sarthak2601** [comment]
> @Sarthak2601 Looks good, thanks.
> Update the title of this PR: `Fix #678: Landscape onBoarding workflow`
> 
> Also, now we can focus on test cases.
> 
> Check `OnboardingFragmentTest` in this you will find a lot of test cases for portrait mode. In this file you will need add following test cases for orientation mode.
> 
> 1. Rotate to orientation mode, check slide_0 title, if this is correct or not.
> 2. In portrait mode, move to slide_1 , change to landscape mode and now check if the title is correct or not.
> 3. In portrait mode, click on skip, change to landscape mode and now check the title is same as last slide title or not.
> 
> Regarding how to change to orientation landscape in a test case, you can see test cases in `AdminAuthActivityTest`

-Added test cases 
-Changed the PR title 

@rt4914, PTAL :-)

File: app/src/sharedTest/java/org/oppia/app/onboarding/OnboardingFragmentTest.kt [hunk 6]
```diff
@@ -437,4 +439,39 @@ class OnboardingFragmentTest {
...
+
+  @Test
+  fun testOnboardingFragment_clickonskip_changeorientation_checktitle() {
+    launch(OnboardingActivity::class.java).use {
+      onView(withId(R.id.skip_text_view)).perform(click())
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.slide_title_text_view))
+        .check(
+          matches(
+            withText(R.string.onboarding_slide_3_title)))
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/onboarding/OnboardingFragmentTest.kt]: Indentation change suggested:
```
      onView(withId(R.id.skip_text_view)).perform(click())
      onView(isRoot()).perform(orientationLandscape())
      onView(withId(R.id.slide_title_text_view)).check(matches(withText(R.string.onboarding_slide_3_title)))
```

Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/app/onboarding/OnboardingFragmentTest.kt]: Ditto for above two cases also.

File: app/src/sharedTest/java/org/oppia/app/onboarding/OnboardingFragmentTest.kt [hunk 7]
```diff
@@ -437,4 +439,39 @@ class OnboardingFragmentTest {
...
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.slide_title_text_view))
+        .check(
+          matches(
+            withText(R.string.onboarding_slide_1_title)))
+    }
+  }
+
+  @Test
+  fun testOnboardingFragment_clickonskip_changeorientation_checktitle() {
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/onboarding/OnboardingFragmentTest.kt]: Test case name:
`testOnboardingFragment_clickOnSkip_changeOrientation_titleIsCorrect`

File: app/src/sharedTest/java/org/oppia/app/onboarding/OnboardingFragmentTest.kt [hunk 8]
```diff
@@ -437,4 +439,39 @@ class OnboardingFragmentTest {
...
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.slide_title_text_view))
+        .check(
+          matches(
+            withText(R.string.onboarding_slide_0_title)))
+    }
+  }
+
+  @Test
+  fun testOnboardingFragment_movetoslide1_changeorientation_checktitle() {
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/onboarding/OnboardingFragmentTest.kt]: Test case name:
`testOnboardingFragment_moveToSlide1_changeOrientation_titleIsCorrect`

File: app/src/sharedTest/java/org/oppia/app/onboarding/OnboardingFragmentTest.kt [hunk 9]
```diff
@@ -437,4 +439,39 @@ class OnboardingFragmentTest {
       ).check(matches(withText(R.string.onboarding_slide_3_title)))
     }
   }
+
+  @Test
+  fun testOnboardingFragment_slide0title_inlandscape_iscorrectornot(){
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/onboarding/OnboardingFragmentTest.kt]: Nit:
`testOnboardingFragment_slide0Title_changeOrientation_titleIsCorrect `

**Review by rt4914 — CHANGES_REQUESTED** [review]
Some small changes suggested.

**Review by rt4914 — CHANGES_REQUESTED** [review]
These 3 test cases are failing.
Update them as following:
```
  @Test
  fun testOnboardingFragment_slide0title_inlandscape_iscorrectornot() {
    launch(OnboardingActivity::class.java).use {
      onView(isRoot()).perform(orientationLandscape())
      onView(
        allOf(
          withId(R.id.slide_title_text_view),
          isCompletelyDisplayed()
        )
      ).check(matches(withText(R.string.onboarding_slide_0_title)))
    }
  }

  @Test
  fun testOnboardingFragment_movetoslide1_changeorientation_checktitle() {
    launch(OnboardingActivity::class.java).use {
      onView(withId(R.id.onboarding_slide_view_pager)).perform(scrollToPage(1))
      onView(isRoot()).perform(orientationLandscape())
      onView(
        allOf(
          withId(R.id.slide_title_text_view),
          isCompletelyDisplayed()
        )
      ).check(matches(withText(R.string.onboarding_slide_1_title)))
    }
  }

  @Test
  fun testOnboardingFragment_clickonskip_changeorientation_checktitle() {
    launch(OnboardingActivity::class.java).use {
      onView(withId(R.id.skip_text_view)).perform(click())
      onView(isRoot()).perform(orientationLandscape())
      onView(
        allOf(
          withId(R.id.slide_title_text_view),
          isCompletelyDisplayed()
        )
      ).check(matches(withText(R.string.onboarding_slide_3_title)))
    }
  }
```

File: app/src/sharedTest/java/org/oppia/app/onboarding/OnboardingFragmentTest.kt (hunk 7 above)
Comment (Sarthak2601) [reply app/src/sharedTest/java/org/oppia/app/onboarding/OnboardingFragmentTest.kt]: Done.

File: app/src/sharedTest/java/org/oppia/app/onboarding/OnboardingFragmentTest.kt (hunk 8 above)
Comment (Sarthak2601) [reply app/src/sharedTest/java/org/oppia/app/onboarding/OnboardingFragmentTest.kt]: Done.

File: app/src/sharedTest/java/org/oppia/app/onboarding/OnboardingFragmentTest.kt (hunk 9 above)
Comment (Sarthak2601) [reply app/src/sharedTest/java/org/oppia/app/onboarding/OnboardingFragmentTest.kt]: Done.

File: app/src/sharedTest/java/org/oppia/app/onboarding/OnboardingFragmentTest.kt (hunk 6 above)
Comment (Sarthak2601) [reply app/src/sharedTest/java/org/oppia/app/onboarding/OnboardingFragmentTest.kt]: > These 3 test cases are failing.
> Update them as following:
> 
> ```
>   @Test
>   fun testOnboardingFragment_slide0title_inlandscape_iscorrectornot() {
>     launch(OnboardingActivity::class.java).use {
>       onView(isRoot()).perform(orientationLandscape())
>       onView(
>         allOf(
>           withId(R.id.slide_title_text_view),
>           isCompletelyDisplayed()
>         )
>       ).check(matches(withText(R.string.onboarding_slide_0_title)))
>     }
>   }
> 
>   @Test
>   fun testOnboardingFragment_movetoslide1_changeorientation_checktitle() {
>     launch(OnboardingActivity::class.java).use {
>       onView(withId(R.id.onboarding_slide_view_pager)).perform(scrollToPage(1))
>       onView(isRoot()).perform(orientationLandscape())
>       onView(
>         allOf(
>           withId(R.id.slide_title_text_view),
>           isCompletelyDisplayed()
>         )
>       ).check(matches(withText(R.string.onboarding_slide_1_title)))
>     }
>   }
> 
>   @Test
>   fun testOnboardingFragment_clickonskip_changeorientation_checktitle() {
>     launch(OnboardingActivity::class.java).use {
>       onView(withId(R.id.skip_text_view)).perform(click())
>       onView(isRoot()).perform(orientationLandscape())
>       onView(
>         allOf(
>           withId(R.id.slide_title_text_view),
>           isCompletelyDisplayed()
>         )
>       ).check(matches(withText(R.string.onboarding_slide_3_title)))
>     }
>   }
> ```

Done.

**Review by rt4914 — APPROVED** [review]
LGTM. Thanks.

**Comment by Sarthak2601** [comment]
> LGTM. Thanks.

Thank you so much ! 
You helped me a lot in this :-)

**Comment by rt4914** [comment]
> > LGTM. Thanks.
> 
> Thank you so much !
> You helped me a lot in this :-)

No worries. I will mostly assign you some new tasks to work on tomorrow.

**Comment by Sarthak2601** [comment]
> > > LGTM. Thanks.
> > 
> > 
> > Thank you so much !
> > You helped me a lot in this :-)
> 
> No worries. I will mostly assign you some new tasks to work on tomorrow.

Sounds great !

**Pull request merged**